spec/requests/schools_request_spec.rb

```diff
@@ -0,0 +1,89 @@
+require 'rails_helper'
+
+RSpec.describe "Schools", type: :request do
+
+  describe "Creating a school" do
+    it "successfully creates a school" do
+      # Given
+      expected_name = "Test School"
+      expected_date = Date.new(2020, 1, 2)
+
+      post "/schools", params: { school:
+                                   { name: expected_name,
+                                     "opened(1i)": expected_date.year.to_s,
+                                     "opened(2i)": expected_date.month.to_s,
+                                     "opened(3i)": expected_date.day.to_s,
+                                   } }
+
+      expect(response).to redirect_to assigns(:school)
+      expect(School.find_by(name: expected_name)).not_to be_nil
+      expect(School.find_by(name: expected_name).opened).to eq(expected_date)
+    end
+
+    it "successfully creates a school with a type" do
+      # Given
+      expected_name = "Test School 2"
+      expected_date = Date.new(2020, 1, 3)
+      expected_school_type = "Primary"
+
+      post "/schools", params: { school:
+                                   {
+                                     name: expected_name,
+                                     "opened(1i)": expected_date.year.to_s,
+                                     "opened(2i)": expected_date.month.to_s,
+                                     "opened(3i)": expected_date.day.to_s,
+                                     school_type: expected_school_type,
+                                   } }
+
+      expect(response).to redirect_to assigns(:school)
+      expect(School.find_by(name: expected_name)).not_to be_nil
+      expect(School.find_by(name: expected_name).opened).to eq(expected_date)
+      expect(School.find_by(name: expected_name).school_type).to eq(expected_school_type)
+    end
+
+    it "will not create a school without a name" do
+      post "/schools", params: { school:
+                                   { name: "",
+                                     "opened(1i)": "2020",
+                                     "opened(2i)": "1",
+                                     "opened(3i)": "1",
+                                   } }
+
+      expect(response).to have_http_status(:success)
+      expect(response).to render_template(:new)
+      expect(assigns(:school).errors.empty?).to be false
```

Comment:
Might also be worth testing the total number of schools to be 0?

Reply:
I would prefer something like 
```
    it "will not create a school without a name" do
      expect do

        post "/schools", params: { school:
                                     { name: "",
                                       "opened(1i)": "2020",
                                       "opened(2i)": "1",
                                       "opened(3i)": "1",
                                     } }

        expect(response).to have_http_status(:success)
        expect(response).to render_template(:new)
        expect(assigns(:school).errors.empty?).to be false
      end.not_to(change{School.count})
    end
```
but it's a really good point that it's something we should test!